src/H5Oattribute.c

```diff
@@ -211,6 +211,13 @@ H5O__attr_create(const H5O_loc_t *loc, H5A_t *attr)
     if (NULL == (oh = H5O_pin(loc)))
         HGOTO_ERROR(H5E_ATTR, H5E_CANTPIN, FAIL, "unable to pin object header");
 
+    /* Check for creating attribute with unusual datatype */
+    if (H5T_is_numeric_with_unusual_unused_bits(attr->shared->dt) &&
```

Comment:
Maybe move the checksum part up so we can short circuit the more expensive function call